docs/source/tutorial.rst

```diff
@@ -7,47 +7,171 @@ Tutorial
+            url="https://zenodo.org/record/1322542/files/cante100_automaticTranscription.zip?download=1",
+            checksum="47fea64c744f9fe678ae5642a8f0ee8e",  # the md5 checksum
+            destination_dir="cante100_automaticTranscription",  # relative path for where to unzip the data, or None
+        ),
+        "metadata": download_utils.RemoteFileMetadata(
+            filename="cante100Meta.xml",
+            url="https://zenodo.org/record/1322542/files/cante100Meta.xml?download=1",
+            checksum="6cce186ce77a06541cdb9f0a671afb46",  # the md5 checksum
+            destination_dir=None,  # relative path for where to unzip the data, or None
+        ),
+        "README": download_utils.RemoteFileMetadata(
+            filename="cante100_README.txt",
+            url="https://zenodo.org/record/1322542/files/cante100_README.txt?download=1",
+            checksum="184209b7e7d816fa603f0c7f481c0aae",  # the md5 checksum
+            destination_dir=None,  # relative path for where to unzip the data, or None
+        ),
+    }
+
+An example for cante100 dataset could be: ``cante100.download(partial_download=['spectrogram', 'melody', 'metadata'])``.
 
 
 Validating a dataset
 ^^^^^^^^^^^^^^^^^^^^
 
-big datasets comment
+Using the method ``validate()`` we can check if the files in the local version are the same than the canical version.
 
+big datasets comment:
+In future ``mirdata`` version, a random validation will be included. This improvement will reduce validation time for very big datasets.
 
 Accessing annotations
 ^^^^^^^^^^^^^^^^^^^^^
-choice track
-select particular track
-annotation classes and compatibility with jams/mir_eval
 
+- Choice track
+We can chose a random track with ``choice_track()`` method.
+
+.. code-block:: python
+
+    random_track = orchset.choice_track()
+
+
+- Select particular track
+
+.. code-block:: python
+
+    orchset_data = orchset.load_tracks()
+    example_track = orchset_data["Beethoven-S3-I-ex1"]
+
+    # Accessing to track melody annotation
+    example_melody = example_track.melody
+
+Annotations can also be accesses through load_ methods.
+
+.. code-block:: python
+
+    random_track = orchset.choice_track()
+
+    # Parsing melody annotation path
+    random_melody_path = random_track.melody_path
+
+    # Accessing to track melody annotation
+    random_melody = orchset.load_melody(melody_path)
+
+- Annotation classes and compatibility with jams/mir_eval
+
+TODO
 
 Iterating over datasets and annotations
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In general, most of the datasets are based on the track concept. Each track has an audio with his own annotations.
+
+With the ``load_tracks()`` method all the tracks (so including their respective audio and annotations) can be loaded as a dictionary structure.
+
+.. code-block:: python
+
+    for key, track in orchset.load_tracks().items():
+        print(key, track.title, track.audio_path)
```

Comment:
I think we can attach both! And then they can chose what works better for them.

Reply:
I like the two examples! I feel it will also be nice to add an example when data is processed just to illustrate a pipeline with `mirdata`, maybe just recycle the example that used to be in the docs?

```
### Evaluate a melody extraction algorithm on Orchset
```python
import mir_eval
import mirdata
import numpy as np
import sox

def very_bad_melody_extractor(audio_path):
    duration = sox.file_info.duration(audio_path)
    time_stamps = np.arange(0, duration, 0.01)
    melody_f0 = np.random.uniform(low=80.0, high=800.0, size=time_stamps.shape)
    return time_stamps, melody_f0

# Evaluate on the full dataset
orchset_scores = {}
orchset = mirdata.Dataset('orchset')
orchset_data = orchset.load_tracks()
for track_id, track_data in orchset_data.items():
    est_times, est_freqs = very_bad_melody_extractor(track_data.audio_path_mono)

    ref_times = track_data.melody.times
    ref_freqs = track_data.melody.frequencies

    score = mir_eval.melody.evaluate(ref_times, ref_freqs, est_times, est_freqs)
    orchset_scores[track_id] = score

# Split the results by composer and by instrumentation
composer_scores = {}
strings_no_strings_scores = {True: {}, False: {}}
for track_id, track_data in orchset_data.items():
    if track_data.composer not in composer_scores.keys():
        composer_scores[track_data.composer] = {}

    composer_scores[track_data.composer][track_id] = orchset_scores[track_id]
    strings_no_strings_scores[track_data.contains_strings][track_id] = \
        orchset_scores[track_id]
```